### ISSUE TITLE
When creating an entry based on a PDF, the file path should be relative

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#11173]
When an entry is created using a PDF (by drag'n'drop on the main table - either at the free space or **between** two lines), the path of the linked PDF is stored as absolute path, but not relative. This needs to be fixed.

Implementation hint: There is functionality inside JabRef to make a path relative. It "just" needs to be found and wired correctly into that drop handling

## Pre condition

Setting that files are stored relative to library file location.

![image](https://github.com/JabRef/jabref/assets/1366654/1c750f8f-8933-492e-91c1-bd909e8fb070)

## Details

![b](https://github.com/JabRef/jabref/assets/1366654/051022fa-2f41-454c-97c1-32d34cbeb167)

Example PDF: [IST2015.pdf](https://github.com/JabRef/jabref/files/14915608/IST2015.pdf)

Created BibTeX:

```bibtex
@TechReport{Steinmachera2014,
  author   = {Igor Steinmachera and b and Marco Aurelio Graciotto Silvaa and Marco Aurelio Gerosab and David F. Redmilesc},
  title    = {A systematic literature review on the barriers faced by newcomers to open source software projects},
  year     = {2014},
  number   = {on},
  file     = {:C\:/Users/olive/OneDrive/Präsentationen/TUG 2023/JabRef/TUG 2023/demo/IST2015.pdf:PDF},
  keywords = {Open Source Software, Software Engineering, Newcomers, Beginners, Novices, Joining, Contribution, Barriers to Entry, Onboarding, Open Collaboration, Socialization, Systematic Literature Review},
}
```

See that the content of `file` has an absolute path. That should be relative.

## Follow-up issue

The newly created entry is not added to the currently selected group. When working on this, please also try to fix this.

## Other places where drag'n'drop works correctly

- Dropping a file onto a line in the main table (leads to attachment instead of creation)
- Dropping a file into the "file" field in the entry editor  (leads to attachment instead of creation)
- Dropping a file into the entry preview (leads to attachment instead of creation)

**Comment by koppor** [comment]
@Arshadpd Seeing the personpower behind this issue, you could try to craft test cases using [TestFX](http://testfx.github.io/TestFX/) - or work on other issues, too.

**Comment by ThiloteE** [comment]
Are you still pursuing this issue?

**Comment by Arshadpd** [comment]
Yes, we are working on this. We have the solution and implementation ready in local, and just trying to check if we need to change any test case for the respective change.

<img width="982" alt="image" src="https://github.com/JabRef/jabref/assets/38530849/3b371b1d-1159-4151-8e5d-84492d54eec0">

The file attribute now, has the relative path. Please let us know if this looks good

**Comment by ThiloteE** [comment]
On my machine with some custom configurations (which should be irrelevant here), using JabRef's development version (Windows 10), a relative file path looks like this, which differs from yours:
![image](https://github.com/JabRef/jabref/assets/73715071/3eaf9488-0554-49bf-a659-f709f1f07490)
![image](https://github.com/JabRef/jabref/assets/73715071/5c802e6e-0fa7-4466-8ec8-4aff1ef591d1)
Unfortunately, I have not found any documentation about relative file path syntax, that would explain the reasoning behind it in more detail. I looked here: https://docs.jabref.org/finding-sorting-and-cleaning-entries/filelinks#directories-for-files.

I see that you have `\` in there, which is a Windows path sign. Have you tested your solution on Linux?

**Comment by Arshadpd** [comment]
I believe this is a different issue from the one reported, we are working on the issue reported by Oliver, where the file path in the BibTex source is not relative. As per the fix that I have in my local, it is now storing the expected relative path in the BibTex Source.

@koppor please let me know if my understanding is correct.

**Comment by Arshadpd** [comment]
On a second thought, @ThiloteE I believe the preferences that you have appears to be different from the one Oliver has provided. The ImportHandler class that we are working on for the issue reported by Oliver, does not seem to be handling the regular expressions in the preferences, it could be a different logic to handle the regex expressions in the preferences.

**Comment by koppor** [comment]
@Arshadpd I see that the file path is relative. 

- I did not see a video for drag'n'drop.
- I also did not see any different test configurations. The provided preference setting, or a setting that inside a library ("user-specific file directoy"). See https://docs.jabref.org/finding-sorting-and-cleaning-entries/filelinks#directories-for-files for details. I
- I did not see any pull request

**Comment by Arshadpd** [comment]
Hi @koppor , Thanks for the pointers,

- I am attaching the zip file of the video below as git doesn't support more than 10 MB.
[Uploading Jabref_Rel_Issue_Fix.zip…]()
- The preferences are also recorded in the video. Please let me know if you need us to run some specific tests.
- Once the video is approved, I will be opening the pull request.

**Comment by ThiloteE** [comment]
By the way, thank you for your efforts and motivation to work on this.

Yes, my preferences are different from what Oliver provided, but it is still something that JabRef supports. If you provide a pull-request with the code, I can do some simple tests myself, such as moving my .bib library to a new location and importing it. Or you copy my preferences and do the tests on your fork, then you can rule out any syntax differences that stem from enabling "linked file name conventions". My intention is simply to check for breaking changes.

The link to the video you provided does not work 🙅

**Comment by Arshadpd** [comment]
Hi @ThiloteE , Sorry for the inconveniences, I'll add the video to my drive and share the link here. I'll be creating a pull request as soon as possible. Thanks for all the support.

**Comment by koppor** [comment]
@Arshadpd You can just use [loom](https://www.loom.com/) to record things

**Comment by Arshadpd** [comment]
https://drive.google.com/file/d/1wG71ME3lfLlR_YcsrRy815onngvlgoyG/view?usp=sharing

Thanks Oliver, I've already screen recorded the functionality in local. Please let me know if the above link is accessible and the thoughts on the video.

**Comment by koppor** [comment]
@Arshadpd I see it. Now I am curious to the code.

**Comment by Arshadpd** [comment]
Thanks @koppor , I have created the pull request, please let me know if the request is correct, or do we need to change something

https://github.com/JabRef/jabref/pull/11327

**Comment by Arshadpd** [comment]
Hi @koppor , Apologies for a little delay on this, but I have resumed working on this, and I have taken the relativize method from "LinkedFilesHandler" class to reuse it in the "ImportHandler". I am attaching the screenshot for your reference, please let me know if this approach is right.
<img width="212" alt="image" src="https://github.com/JabRef/jabref/assets/38530849/88cfa954-c169-493c-8780-eb3f00dbc761">
<img width="541" alt="image" src="https://github.com/JabRef/jabref/assets/38530849/85b0a8fd-c3fa-4386-8851-eeec9a1f71db">

**Comment by koppor** [comment]
> Hi @koppor , Apologies for a little delay on this, but I have resumed working on this, and I have taken the relativize method from "LinkedFilesHandler" class to reuse it in the "ImportHandler". I am attaching the screenshot for your reference, please let me know if this approach is right.
> <img width="212" alt="image" src="https://github.com/JabRef/jabref/assets/38530849/88cfa954-c169-493c-8780-eb3f00dbc761">
> <img width="541" alt="image" src="https://github.com/JabRef/jabref/assets/38530849/85b0a8fd-c3fa-4386-8851-eeec9a1f71db">
> 

Looks good 👍

Note that you can use Markdown to paste JavaCode. (Enclosed in three backticks)

**Comment by Arshadpd** [comment]
Hi Oliver, I am trying to write test cases for this code, just wanted to know, if there is any exiting method that I need to add logic to. 

I have looked in ImportHabndlerTest.java, but it does not have the code to test importFilesinBackground methods, existing logic.

**Comment by Arshadpd** [comment]
Hi @koppor , just wanted to follow up on the test case advise, can you please guide us on this, I have the code working and ready to be pushed. Please let us know what should be the next step if the existing code does not have test cases and is it possible to create the pull request without that.

**Comment by Techsavy3000** [comment]
Hi @koppor We have attached the test cases document for your review. Looking forward to your feedback.
[Test Case Report - ( # 11173).pdf](https://github.com/user-attachments/files/15908233/Test.Case.Report.-.11173.pdf)

**Comment by Sourav-Debnath** [comment]
Hello @koppor we have made an UML class diagram and report on the modification related with the solution of the issue. Can you please review and give us your feedback.
[Technical Report on Modifications to the ImportHandler Class[#11173].pdf](https://github.com/user-attachments/files/15908300/Technical.Report.on.Modifications.to.the.ImportHandler.Class.11173.pdf)

**Comment by Arshadpd** [comment]
Hi @koppor , We have sent another pull request, with the updated code that you approved earlier in the comments above. Please let us know your thoughts on the same.

**Comment by koppor** [comment]
Test plan was very strange. No UI test mentioned. No evaluation of TestFX.

**Comment by devangshetty** [comment]
Initial Setup and challenges that we faced
[JabRef-Setup.pdf](https://github.com/user-attachments/files/15935160/JabRef-Setup.pdf)

**Comment by koppor** [comment]
> Initial Setup and challenges that we faced [JabRef-Setup.pdf](https://github.com/user-attachments/files/15935422/JabRef-Setup.pdf)

Thank you! Seeing that you needed to learn git, it seems that this project was too large and required too much upfront knowledge. I checked our devdocs. We do say: Use Gradle in the IDE. However, there are two or three places where we refer to `./gradlew` and do not state how to setup a JDK. We will try to rework that and state: Always use IntelliJ 😅
